services/cd-service/pkg/repository/overview_cache_test.go

```diff
@@ -0,0 +1,438 @@
+				{
+					Applications: map[string]*api.Environment_Application{
+						"bar": { // different app
+							UndeployVersion: true,
+							Version:         666,
+						},
+					},
+				},
+			}),
+			ExpectedResult: api.UndeploySummary_UNDEPLOY,
+		},
+		{
+			Name:    "One Env with undeploy",
+			AppName: "foo",
+			groups: groupFromEnvs([]*api.Environment{
+				{
+					Applications: map[string]*api.Environment_Application{
+						"foo": {
+							UndeployVersion: true,
+							Version:         666,
+						},
+					},
+				},
+			}),
+			ExpectedResult: api.UndeploySummary_UNDEPLOY,
+		},
+		{
+			Name:    "One Env with normal version",
+			AppName: "foo",
+			groups: groupFromEnvs([]*api.Environment{
+				{
+					Applications: map[string]*api.Environment_Application{
+						"foo": {
+							UndeployVersion: false,
+							Version:         666,
+						},
+					},
+				},
+			}),
+			ExpectedResult: api.UndeploySummary_NORMAL,
+		},
+		{
+			Name:    "Two Envs all undeploy",
+			AppName: "foo",
+			groups: groupFromEnvs([]*api.Environment{
+				{
+					Applications: map[string]*api.Environment_Application{
+						"foo": {
+							UndeployVersion: true,
+							Version:         666,
+						},
+					},
+				},
+				{
+					Applications: map[string]*api.Environment_Application{
+						"foo": {
+							UndeployVersion: true,
+							Version:         666,
+						},
+					},
+				},
+			}),
+			ExpectedResult: api.UndeploySummary_UNDEPLOY,
+		},
+		{
+			Name:    "Two Envs all normal",
+			AppName: "foo",
+			groups: groupFromEnvs([]*api.Environment{
+				{
+					Applications: map[string]*api.Environment_Application{
+						"foo": {
+							UndeployVersion: false,
+							Version:         666,
+						},
+					},
+				},
+				{
+					Applications: map[string]*api.Environment_Application{
+						"foo": {
+							UndeployVersion: false,
+							Version:         666,
+						},
+					},
+				},
+			}),
+			ExpectedResult: api.UndeploySummary_NORMAL,
+		},
+		{
+			Name:    "Two Envs all different",
+			AppName: "foo",
+			groups: groupFromEnvs([]*api.Environment{
+				{
+					Applications: map[string]*api.Environment_Application{
+						"foo": {
+							UndeployVersion: true,
+							Version:         666,
+						},
+					},
+				},
+				{
+					Applications: map[string]*api.Environment_Application{
+						"foo": {
+							UndeployVersion: false,
+							Version:         666,
+						},
+					},
+				},
+			}),
+			ExpectedResult: api.UndeploySummary_MIXED,
+		},
+	}
+	for _, tc := range tcs {
+		t.Run(tc.Name, func(t *testing.T) {
+			actualResult := DeriveUndeploySummary(tc.AppName, tc.groups)
+			if !cmp.Equal(tc.ExpectedResult, actualResult) {
+				t.Fatal("Output mismatch (-want +got):\n", cmp.Diff(tc.ExpectedResult, actualResult))
+			}
+		})
+	}
+}
+
+func TestUpdateOverviewCacheDB(t *testing.T) {
+	tcs := []struct {
+		Name           string
+		CreateTestData func(context.Context, *sql.Tx, *db.DBHandler) error
```

Comment:
having a function like this in a test array indicates that these should be either
A) Separate Tests
B) the data the function produces should instead be stored in the test array directly.
Either way, there should not be function pointers in a test array.